### DISCUSSION TITLE
Where is compile listing?

### DISCUSSION THREAD

**Discussion by SteversJE** [codefori/vscode-ibmi#2305] [Q&A] [answered]
I can't find the compile listing on the IBM box after the object is created. Where does it go? I sometimes compile something so I know where to look when I see "MCH0603 unmonitored by UXINVOICE at statement 0000292300." Is there an alternative to the compile listing?

**Comment by thomprl** [comment]
If you are compiling inside of VSCode.  The compile listing usually shows on the terminal tab at the bottom of the screen.  I'm pretty sure VSCode converts the spool file to text and then loads it into the terminal tab and deletes the spool file.  Then you can use the problems tab to see any errors.  

![image](https://github.com/user-attachments/assets/71fb48f1-15f6-4df2-abdd-503fa29a03e7)

**Reply by SteversJE** [reply]
I understood that part of it. This just addresses the immediate use of the compile listing for the compile-edit-debug cycle, not my longer term (months) use of keeping it around for production troubleshooting.

**Comment by sebjulliand** [comment]
Since every command executed by Code for i are run into a shell, the output doesn't go into a spooled file but directly in the shell's standard output. That output is logged in the terminal used to run the action (as @thomprl said).

**Reply by thomprl** [reply]
Thanks, wasn't sure about the magic behind the scenes.

**Reply by SteversJE** [reply]
(I can't tell if I'm replying to sebjulliand or thomprl; I'll find out after I post.)

Thanks sebjulliand. I guess the implication is I can't change the behavior of the extension to keep the compile listing, but I can get my desired outcome by creating additional commands to submit the compiles to batch. (Which I already tested and does work fine.)

**Reply by chrjorgensen** [reply, marked as answer]
@SteversJE Instead of keeping the spoolfile from the compile, you should compile it into the program object using DBGVIEW *ALL or *LISTING.

That way you always have the compile listing at hand, it follows the program into production. 

You see it when you STRDBG against the program.

**Reply by SteversJE** [reply]
Thanks. I'm looking at that *LIST approach. I just hope the line is easy to find and doesn't require scrolling and perusing.

(Okay, I'm going off on a tangent here. Thread can be done.)
